lib/ansible/modules/network/avi/avi_stringgroup.py

```diff
@@ -0,0 +1,135 @@
+#!/usr/bin/python
+#
+# Created on Aug 25, 2016
+# @author: Gaurav Rastogi (grastogi@avinetworks.com)
+#          Eric Anderson (eanderson@avinetworks.com)
+# module_check: supported
+# Avi Version: 17.1.1
+#
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+ANSIBLE_METADATA = {'metadata_version': '1.0',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: avi_stringgroup
+author: Gaurav Rastogi (grastogi@avinetworks.com)
+
+short_description: Module for setup of StringGroup Avi RESTful Object
+description:
+    - This module is used to configure StringGroup object
+    - more examples at U(https://github.com/avinetworks/devops)
+requirements: [ avisdk ]
+version_added: "2.4"
+options:
+    state:
+        description:
+            - The state that should be applied on the entity.
+        default: present
+        choices: ["absent","present"]
+    description:
+        description:
+            - User defined description for the object.
+    kv:
+        description:
+            - Configure key value in the string group.
+    name:
+        description:
+            - Name of the string group.
+        required: true
+    tenant_ref:
+        description:
+            - It is a reference to an object of type tenant.
+    type:
+        description:
+            - Type of stringgroup.
+            - Enum options - SG_TYPE_STRING, SG_TYPE_KEYVAL.
```

Comment:
yes, these are the only valid options but if controller is upgraded to newer builds then using choices would be limiting. It would make modules use a bit more restrictive. I would propose we don't move it into the choices.